src/PoshGitTypes.ps1

```diff
@@ -0,0 +1,213 @@
+            if ($this.CustomAnsi) {
+                $e = [char]27 + "["
+                $ansi = $this.CustomAnsi
+                $escAnsi = EscapseAnsiString $this.CustomAnsi
+                $txt = $this.RenderAnsi()
+                $str = "Text: '$txt',`t CustomAnsi: '${ansi}${escAnsi}${e}0m'"
+            }
+            else {
+                $color = [PoshGitCellColor]::new($this.ForegroundColor, $this.BackgroundColor)
+                $txt = $this.RenderAnsi()
+                $str = "Text: '$txt',`t $($color.ToString())"
+            }
+        }
+        else {
+            $color = [PoshGitCellColor]::new($this.ForegroundColor, $this.BackgroundColor)
+            $txt = $this.Text
+            $str = "Text: '$txt',`t $($color.ToString())"
+        }
+
+        return $str
+    }
+
+    [string] RenderAnsi() {
+        $e = [char]27 + "["
+        $txt = $this.Text
+
+        if ($this.CustomAnsi) {
+            $ansi = $this.CustomAnsi
+            $str = "${ansi}${txt}${e}0m"
+        }
+        else {
+            $bg = $this.BackgroundColor
+            if ($bg -and !(Test-VirtualTerminalSequece $bg)) {
+                $bg = Get-BackgroundVirtualTerminalSequence $bg
+            }
+
+            $fg = $this.ForegroundColor
+            if ($fg -and !(Test-VirtualTerminalSequece $fg)) {
+                $fg = Get-ForegroundVirtualTerminalSequence $fg
+            }
+
+            $str = "${fg}${bg}${txt}${e}0m"
+        }
+
+        return $str
+    }
+}
+
+class GitPromptSettings {
+    [bool]$AnsiConsole = $Host.UI.SupportsVirtualTerminal -or ($Env:ConEmuANSI -eq "ON")
+
+    [PoshGitCellColor]$DefaultColor = [PoshGitCellColor]::new()
+    [PoshGitCellColor]$BranchColor = [PoshGitCellColor]::new([ConsoleColor]::Cyan)
+
+    [PoshGitCellColor]$IndexColor = [PoshGitCellColor]::new([ConsoleColor]::DarkGreen)
+    [PoshGitCellColor]$WorkingColor = [PoshGitCellColor]::new([ConsoleColor]::DarkRed)
+    [PoshGitCellColor]$StashColor = [PoshGitCellColor]::new([ConsoleColor]::Red)
+    [PoshGitCellColor]$ErrorColor = [PoshGitCellColor]::new([ConsoleColor]::Red)
+
+    [PoshGitTextSpan]$BeforeText = [PoshGitTextSpan]::new(' [', [ConsoleColor]::Yellow)
+    [PoshGitTextSpan]$DelimText = [PoshGitTextSpan]::new(' |', [ConsoleColor]::Yellow)
+    [PoshGitTextSpan]$AfterText = [PoshGitTextSpan]::new(']', [ConsoleColor]::Yellow)
+
+    [PoshGitTextSpan]$BeforeIndexText = [PoshGitTextSpan]::new('', [ConsoleColor]::DarkGreen)
+    [PoshGitTextSpan]$BeforeStashText = [PoshGitTextSpan]::new(' (', [ConsoleColor]::Red)
+    [PoshGitTextSpan]$AfterStashText = [PoshGitTextSpan]::new(')', [ConsoleColor]::Red)
+
+    [PoshGitTextSpan]$LocalDefaultStatusSymbol = [PoshGitTextSpan]::new('', [ConsoleColor]::DarkGreen)
+    [PoshGitTextSpan]$LocalWorkingStatusSymbol = [PoshGitTextSpan]::new('!', [ConsoleColor]::DarkRed)
+    [PoshGitTextSpan]$LocalStagedStatusSymbol = [PoshGitTextSpan]::new('~', [ConsoleColor]::DarkCyan)
```

Comment:
In case you missed the comment on an outdated line: was this deliberately changed from `Cyan` to `DarkCyan`?

Reply:
Looks like that was in the very first commit I made back on 10/28.  I honestly don't remember.  I wonder if my color scheme was having issues with Cyan that day?  In my current scheme (OneHalfDark) I can't see any diff between Cyan and DarkCyan.  Want me to change it back to `Cyan`?

Reply:
Done.